roles/repo_setup/tasks/configure.yml

```diff
@@ -5,6 +5,7 @@
   when:
     - content_provider_dlrn_md5_hash is defined
     - content_provider_dlrn_md5_hash | length > 0
+    - (not cifmw_run_update|default(false)) or (update_playbook_run is defined and cifmw_run_update|default(false))
```

Comment:
its a bit confusing

if you are not running an update OR
if you ARE running update AND the update playbook is defined
?
like that?

but what is the case that you are running an update and the update playbook is not defined?

Reply:
cifmw_run_update - is defined on job level, meaning this job is testing update. In that case during deployment phase don't use images from content_provider, but stable/promoted containers instead. 

update_playbook_run is defined - when the role is triggered from within ci/update_playbook, if content_provider is available use during update those containers

Hope it helps

Reply:
So 

 | run_update | update_playbook_run | result | comment |
 | ----- | ---- | ----- | ---- |
| undefined or false | undefined | true: use content provider if defined as usual | the normal deployment where those variables are just not there, so nothing change|
| true | undefined | false: don't use content provider during deployment | a job that is an update job, but running the deployment playbook |
| true | defined | true : use content provider during update | a job that is an update job and is running the update playbook |

It definitively looks correct to me.